src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseStream.cs

```diff
@@ -196,6 +196,11 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
                 throw new InvalidOperationException(SR.net_http_no_concurrent_io_allowed);
             }
 
+            if (count == 0)
+            {
+                return Task.FromResult<int>(0);
+            }
```

Comment:
Nit: as a minor improvement, I'd suggest moving this into ReadAsyncCore and having it be the very first thing done there, e.g.
```C#
private async Task<int> ReadAsyncCore(...)
{
    if (count == 0)
    {
        return count;
    }
    ...
}
```
Two advantages to that:
1) If anything else ends up calling ReadAsyncCore, it'll pick up the same fix.
2) The async infrastructure has a cached task for Int32 0, so whereas this will allocate a task, moving it inside of the ReadAsyncCore will avoid that allocation.